include/lsst/afw/table/Catalog.h

```diff
@@ -760,14 +761,16 @@ typename CatalogT<RecordT>::iterator CatalogT<RecordT>::find(typename Field<T>::
     detail::KeyExtractionFunctor<RecordT, T> f = {key};
     // Iterator adaptor that makes a CatalogT iterator work like an iterator over field values.
     typedef boost::transform_iterator<detail::KeyExtractionFunctor<RecordT, T>, iterator> SearchIter;
-    // Try binary search for log n search assuming the table is sorted.
-    // If the search is unsuccessful, try a brute-force search before quitting.
+    /* Try binary search for log n search assuming the table is sorted.
+     * If the search is unsuccessful, try a brute-force search before quitting.
+     */
     SearchIter i = std::lower_bound(SearchIter(begin(), f), SearchIter(end(), f), value);
-    if (i.base() == end() || *i != value)
-    {
+    if (i.base() == end() || *i != value) {
         i = std::find(SearchIter(begin(), f), SearchIter(end(), f), value);
-        if (i.base() == end() || *i != value)
+        if (i.base() == end()) {
             return end();
+        }
+        LOGL_INFO("afw.table.Catalog", "Catalog is not sorted by the key. Finding a record may be slow.");
```

Comment:
One of the motivations for logging this at some low level is that this is (probably?) the only direct indication we have to let us know that the catalog is unsorted and I think it'd be useful to keep that at TRACE level at least.